conference.js

```diff
@@ -1512,5 +1514,67 @@ export default {
             // Update the view
             APP.UI.setLocalRaisedHandStatus(raisedHand);
         }
+    },
+    
+    /**
+     * When there is anonymous domain enabled together with JWT authentication
+     * binds window message listener, which will wait for JWT token to be
+     * received from the login service opened in a popup window.
+     * @param options config.js
+     * @private
+     */
+    _initJWTTokenListener(options) {
+        if (!options.tokenAuthUrl) {
+            return;
+        }
+        var self = this;
+        var listener = function (event) {
+            if (event.data && event.data.jwtToken) {
+                // FIXME implement origin verification ?
+                //if (event.origin !== window.location.origin) {
+                //    logger.warn("Ignoring JWT token from different origin: " +
+                //        event.origin);
+                //    return;
+                //}
+                config.token = event.data.jwtToken;
+                console.info("Received JWT token:", config.token);
+                var roomName = options.roomName;
+                openConnection({retry: false, roomName: roomName })
+                    .then(function (connection) {
+                        // Start new connection
+                        let newRoom = connection.initJitsiConference(
+                            roomName, self._getConferenceOptions());
+                        // Authenticate from the new connection to get
+                        // the session-ID from the focus, which wil then be used
+                        // to upgrade current connection's user role
+                        newRoom.room.moderator.authenticate().then(function () {
```

Comment:
Aren't we accessing private property of lib-jitsi-meet that way? I think we should implement JitsiConference.authenticate method.